src/main/java/seedu/address/wrapper/UserInfoRoute.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.wrapper;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.awt.image.BufferedImage;
+import java.io.IOException;
+import javax.imageio.ImageIO;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import kong.unirest.Unirest;
+import kong.unirest.UnirestInstance;
+import seedu.address.storage.Storage;
+import seedu.address.wrapper.exceptions.ResponseParseException;
+
+/**
+ * Class representing routes taken to get user information
+ */
+public final class UserInfoRoute {
+    //@@author arnav-ag
+
+    public static final String BASE_GITHUB_URL = "https://api.github.com";
+    private static final String SERVER_DOWN_STATUS_MESSAGE = "Unable to get information from API.";
+
+    private static final String GET_USER_BASE_PATH = "/users/";
+    private final String path;
+    private final Storage storage;
+    private final String username;
+
+    private UserInfoRoute(String path, String username, Storage storage) {
+        requireAllNonNull(path, username, storage);
+        this.path = path;
+        this.storage = storage;
+        this.username = username;
+    }
+
+    public static UserInfoRoute getUserInfoRoute(String username, Storage storage) {
+        requireAllNonNull(username);
+        return new UserInfoRoute(GET_USER_BASE_PATH + username, username, storage);
+    }
+
+    /**
+     * @param unirest Unirest instance to be used to carry out requests
+     * @return Request class pertaining to user information from GitHub
+     */
+    public UserInfoRequest createRequest(UnirestInstance unirest) {
+        requireAllNonNull(unirest);
```

Comment:
I think assertion is a more appropriate error handling here as it likely to be programmer's error.